### DISCUSSION TITLE
Usd import failed in Maya : wrong renderingToSceneFaceVtxIds

### DISCUSSION THREAD

**Discussion by raphaeljnwave** [Autodesk/maya-usd#2405] [Q&A]
Hello maya-usd team,

I faced a strange error while importing a usd file in Maya:
```
// Error: Invalid Hydra prim '/Proxy_asset_to_usd_failed_new_uvShape_0x1c377ba0/myMesh': requires 854 vertices, while the number of elements in renderingToSceneFaceVtxIds is 826. Skipping primvar update. // 
// Error: Invalid Hydra prim '/Proxy_asset_to_usd_failed_new_uvShape_0x1c377ba0/myMesh': requires 854 vertices, while the number of elements in renderingToSceneFaceVtxIds is 826. Skipping primvar update. // 
```
Meanwhile, my usd stage seems correctly imported in Maya but the mesh is not visible in the Maya viewport. However, it becomes visible once rendered (using Arnold for ex) or by switching the maya viewport to GL hydra.

When opening the usd in usdview, I can see that I indeed have 854 vertices as displayed in the error message. However, I have no clue about the number 826 from that same message.

In the following archive you will find the usd file, as well as the maya scene with the mesh that was exported in usd file.

[asset_to_usd_failed.zip](https://github.com/Autodesk/maya-usd/files/8868375/asset_to_usd_failed.zip)

I am currently working with maya2022, python3 and MayaUSD0.18.0.

Do you have any clue on how to visualize correctly that mesh in the maya viewport ?

Thank you !
Raphael

**Comment by santosd** [comment]
Hi @raphaeljnwave I tested this out on my end in Maya 2023 and I am also seeing the same error. I am looking into and will get back to you with more information.

**Comment by williamkrick** [comment]
Hi @raphaeljnwave, In assert_to_usd_failed.usda the primvar myMesh/faceVertexIndices is storing the index of each vertex in a face in the points[] primvar. faceVertexIndices has an entry of 826, which means there must be at least 826 entries in points[]. However, there are only 854 entries in points.

Essentially, faceVertexIndices and faceVertexCounts describe a face that uses vertex 825 and 826, but those vertices don't exist. I'm not sure what usdview does in this case. I'm assuming you can see the mesh in usdview, so possibly it is dropping the bad face from the geometry?

**Comment by raphaeljnwave** [comment]
Hey,
Thanks for you help on that. @williamkrick I am I little bit confused with your numbers.
Here is what my usd contains:
- faceVertexCounts : int[824] (only faces with 4 vertices)
- faceVertexIndices : int[3296] (indeed, 4*824=3296 and max(faceVertexIndices) =853 )
- points : point3f[854] 
- primvar:st : textCoord2f[1125] 
- primvar:st:indices : int[3296] ( max(primvar:st:indices) = 1124 )

These values match the ones from usdview and they have been checked by parsing the usd file (using python).

All seems good at the end. I don't understand where 826 comes from.

**Comment by williamkrick** [comment]
Hi @raphaeljnwave, if you open asset_to_usd_failed.usda that is attached to the original discussion post and search for the text "826" then you'll see that faceVertexIndices does have 826 as a value. Whatever you are doing to parse the file must also be filtering out the faces that use out of range vertex ids.

**Comment by raphaeljnwave** [comment]
Indeed we have 826 in faceVertexIndices, as we have values from [0;853]. But if I well understand, it is correct since it refers to points indices and my points list contains 854 tuples ( and 826 <= 853 ). So for me it seems good. Or maybe I miss something.

**Comment by richardssam** [comment]
I'm seeing the same errors, it essentially is flagging models where there are unused vertices. I'm seeing this generated from FBX files (possibly from Revit), where the USD file was written out by Maya. It feels like if maya is going to flag it, that it perhaps shouldn't be writing it out?
